controllers/main.js

```diff
@@ -22,20 +22,42 @@ const getUrl = async (req, res) => {
     }
     res.json(url);
   } catch (error) {
-    res.status(HTTP_ERRORS.INTERNAL_SERVER_ERROR).json({ error: 'Something went wrong. Try Again' });
+    res
+      .status(HTTP_ERRORS.INTERNAL_SERVER_ERROR)
+      .json({ error: 'Something went wrong. Try Again' });
   }
 };
 
 const editUrl = async (req, res) => {
-  const { originalUrl, customName, shortUrl } = req.body;
+  const { originalUrl, customName } = req.body;
   const { id } = req.params;
   const filter = { _id: id };
-  const url = await Url.findOneAndUpdate(
-    filter,
-    { originalUrl, customName, shortUrl },
-    { new: true }
-  );
-  res.json(url);
+  let shortUrls;
+
+  try {
+    if (customName) {
+      const customNameWithDashes = customName.split(' ').join('-');
+      shortUrls = `${BASEURL}/${customNameWithDashes}`;
+    }
+
+    const url = await Url.findOneAndUpdate(
+      filter,
+      { customName, shortUrl: shortUrls },
+      { new: true, runValidators: true }
+    );
+
+    if (!originalUrl) {
+      return res.status(HTTP_ERRORS.OK).json(url);
+    }
```

Comment:
What does this block do? Because it doesn't make sense to me.

Reply:
Okay, so funny story, based on your requirements, none of the fields should be required but when i didn't add the original url, it gave me a weird error but this block helps combat it as it adds the rest of the fields without the original url. This is another reason why i didn't add the originalUrl to the update method. All i'm going to say, is it right? probably not. Does it meet your requirements and give me the required outputs? i guess it does and that was alright for me

Reply:
This looks like a superficial fix for an error that shouldn't be happening. Remove it. 😅

Reply:
done on 46976ba